ir/json_loader.h

```diff
@@ -331,41 +331,34 @@ class JSONLoader {
         if (auto *s = json->to<JsonString>()) s->c_str() >> v;
     }
 
-    void unpack_json(UnparsedConstant *&v) {
-        cstring text("");
-        unsigned skip = 0;
-        unsigned base = 0;
-        bool hasWidth = false;
-
-        load("text", text);
-        load("skip", skip);
-        load("base", base);
-        load("hasWidth", hasWidth);
-
-        v = new UnparsedConstant({text, skip, base, hasWidth});
-    }
-
     template <typename T>
-    std::enable_if_t<has_fromJSON<T>::value && !std::is_base_of_v<IR::Node, T> &&
+    std::enable_if_t<has_fromJSON<T>::value && !std::is_base_of_v<IR::INode, T> &&
                      std::is_pointer_v<decltype(T::fromJSON(std::declval<JSONLoader &>()))>>
     unpack_json(T *&v) {
         v = T::fromJSON(*this);
     }
 
     template <typename T>
-    std::enable_if_t<has_fromJSON<T>::value && !std::is_base_of_v<IR::Node, T> &&
+    std::enable_if_t<has_fromJSON<T>::value && !std::is_base_of_v<IR::INode, T> &&
                      std::is_pointer_v<decltype(T::fromJSON(std::declval<JSONLoader &>()))>>
     unpack_json(T &v) {
         v = *(T::fromJSON(*this));
```

Comment:
Does it make sense to force move semantics here?

Reply:
You already get move semantics, as T::fromJSON returns a type T xvalue.  That's actually how calling `std::move` forces move semantics -- `std::move` is just a function that returns a type T xvalue.  There's nothing special about it.

Reply:
It can either return a `T` or a `T *`.  If it is a `T *` then it might not be safe to move?